### ISSUE TITLE
Document for the interfaces, classes and methods of workflow-core

### ISSUE THREAD

**Issue by georgelichen** [danielgerlag/workflow-core#341]
Workflow-core is a great package.  Thank you authors!  It would be much helpful, if documents can be provided for the interfaces, classes, methods (with parameters and return values) that the workflow core exposes to the consumer applications.

**Comment by danielgerlag** [comment]
Yes, I agree... I need help with this, my focus is on another project right now... but I will prioritize this afterwards.
